src/main/java/org/takes/rq/multipart/RqMtFake.java

```diff
@@ -143,49 +147,60 @@ public InputStream body() throws IOException {
      * Fake body .
      * @since 0.33
      */
+    @RequiredArgsConstructor
     private static final class FakeBody implements Body {
 
         /**
-         * Content of the body.
+         * The parts.
          */
-        private final String content;
+        private final Request[] parts;
 
         /**
-         * Ctor.
-         * @param parts Fake request body partsFakeBody
-         * @throws IOException If fails
+         * The content.
+         */
+        private final AtomicReference<String> content = new AtomicReference<>();
+
+        /**
+         * The update function.
+         * @checkstyle AnonInnerLengthCheck (100 lines)
          */
-        @SuppressWarnings(
-            "PMD.ConstructorOnlyInitializesOrCallOtherConstructors"
-        )
-        private FakeBody(final Request... parts) throws IOException {
-            final StringBuilder builder = new StringBuilder(128);
-            for (final Request part : parts) {
-                builder.append(String.format("--%s", RqMtFake.BOUNDARY))
-                    .append(RqMtFake.CRLF)
-                    .append("Content-Disposition: ")
-                    .append(
-                        new RqHeaders.Smart(
-                            new RqHeaders.Base(part)
-                        ).single("Content-Disposition")
-                    ).append(RqMtFake.CRLF);
-                final String body = new RqPrint(part).printBody();
-                if (!(RqMtFake.CRLF.equals(body) || body.isEmpty())) {
-                    builder.append(RqMtFake.CRLF)
-                        .append(body)
-                        .append(RqMtFake.CRLF);
+        private final UnaryOperator<String> update = new UnaryOperator<String>() {
```

Comment:
@andreoss WDYT about declaring a private inner class `UpdateUnaryOperator` and instantiate variable `update` in constructor ?

Reply:
@baudoliver7 It would be a second layer of `private innter classes
I just replaced `Atomic*` with `Sticky`  & lambda in ctor.
Here is the discussion which pushed me to use `Atomic*` in the first place https://github.com/yegor256/cactoos/issues/1438.

Reply:
@andreoss Good alternative, thx :)